### PR TITLE
Fix dashboard endpoint - store json blob directly and retrieve it

### PR DESCRIPTION
**Pull request by gokulavasan**
Bamboo : https://builds.cask.co/browse/CDAP-DUT408

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java [hunk 1]
```diff
@@ -60,7 +58,8 @@ public DashboardHttpHandler(Authenticator authenticator) {
   public synchronized void create(final HttpRequest request, final HttpResponder responder,
                                   @PathParam("namespace-id") String namespace) throws Exception {
     String dashboardId = UUID.randomUUID().toString();
-    configStore.put(namespace, dashboardId, GSON.toJson(decodeArguments(request)));
+    String body = request.getContent().toString(Charsets.UTF_8);
```
Comment (abaranau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java]: is the contract that we store String? or any binary

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/DashboardHttpHandlerTest.java [hunk 2]
```diff
@@ -80,13 +80,16 @@ public void testMultiNamespace() throws Exception {
 
   @Test
   public void testProperties() throws Exception {
```
Comment (abaranau) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/DashboardHttpHandlerTest.java]: do you want to test that it stores any String?

**Comment by abaranau** [comment]
couple comments

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/DashboardHttpHandlerTest.java (hunk 2 above)
Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/DashboardHttpHandlerTest.java]: Added a test for the same.

**Comment by gokulavasan** [comment]
Now, returning the id of the newly created dashboard as a json with property "id".

**Comment by gokulavasan** [comment]
@abaranau Please take a look again.

**Comment by abaranau** [comment]
LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java [hunk 3]
```diff
@@ -59,17 +63,32 @@ public DashboardHttpHandler(Authenticator authenticator) {
   @POST
   public synchronized void create(final HttpRequest request, final HttpResponder responder,
                                   @PathParam("namespace-id") String namespace) throws Exception {
+    String data = request.getContent().toString(Charsets.UTF_8);
+    if (!data.equals("") && !isValidJSON(data)) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Invalid Json");
```
Comment (abaranau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java]: "Invalid JSON in body"

**Comment by abaranau** [comment]
LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java [hunk 4]
```diff
@@ -59,17 +63,32 @@ public DashboardHttpHandler(Authenticator authenticator) {
...
+      return;
+    }
+
+    Map<String, String> jsonMap = Maps.newHashMap();
     String dashboardId = UUID.randomUUID().toString();
-    configStore.put(namespace, dashboardId, GSON.toJson(decodeArguments(request)));
-    responder.sendString(HttpResponseStatus.OK, dashboardId);
+    configStore.put(namespace, dashboardId, request.getContent().array());
```
Comment (abaranau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java]: never use .array() on ByteBuffer

**Comment by abaranau** [comment]
fix and merge

**Pull request merged**